### PR TITLE
Remove override to not misuse ProfileKeyedServiceFactory.

### PR DESCRIPTION
**Pull request by goodov**
Resolves https://github.com/brave/brave-browser/issues/41788

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: chromium_src/chrome/browser/themes/theme_service_factory.cc [hunk 1]
```diff
@@ -20,11 +16,5 @@
 #endif
 
 #include "src/chrome/browser/themes/theme_service_factory.cc"
-#undef BRAVE_THEMESERVICEFACTORY_BUILDSERVICEINSTANCEFOR
 
-content::BrowserContext* ThemeServiceFactory::GetBrowserContextToUse(
-    content::BrowserContext* context) const {
-  if (context->IsTor())
-    return context;
```
Comment (goodov) [inline chromium_src/chrome/browser/themes/theme_service_factory.cc:28]: doesn't look like this override is needed now. Tor windows look correct.

this change was introduced here https://github.com/brave/brave-core/pull/6912

File: chromium_src/chrome/browser/search_engines/template_url_service_factory.cc [hunk 2]
```diff
@@ -5,13 +5,11 @@
 
 #include "chrome/browser/search_engines/template_url_service_factory.h"
 
-#include "chrome/browser/history/history_service_factory.h"
-#include "chrome/browser/profiles/incognito_helpers.h"
-#include "chrome/browser/webdata_services/web_data_service_factory.h"
+#define WithAshInternals(...)                      \
+  WithAshInternals(__VA_ARGS__)                    \
+      .WithRegular(ProfileSelection::kOwnInstance) \
```
Comment (goodov) [inline chromium_src/chrome/browser/search_engines/template_url_service_factory.cc:10]: this is covered with tests in `//brave/browser/search_engines/search_engine_provider_service_browsertest.cc`.

File: chromium_src/chrome/browser/search_engine_choice/search_engine_choice_service_factory.cc [hunk 3]
```diff
@@ -5,17 +5,11 @@
 
 #include "chrome/browser/search_engine_choice/search_engine_choice_service_factory.h"
 
-#include "chrome/browser/profiles/incognito_helpers.h"
+#define WithAshInternals(...)                      \
+  WithAshInternals(__VA_ARGS__)                    \
+      .WithRegular(ProfileSelection::kOwnInstance) \
```
Comment (goodov) [inline chromium_src/chrome/browser/search_engine_choice/search_engine_choice_service_factory.cc:10]: also covered with `//brave/browser/search_engines/search_engine_provider_service_browsertest.cc`.

File: chromium_src/chrome/browser/safe_browsing/extension_telemetry/extension_telemetry_service_factory.cc [hunk 4]
```diff
@@ -31,15 +31,9 @@ ExtensionTelemetryServiceFactory::ExtensionTelemetryServiceFactory()
                                  ProfileSelections::BuildNoProfilesSelected()) {
 }
 
-content::BrowserContext*
-ExtensionTelemetryServiceFactory::GetBrowserContextToUse(
-    content::BrowserContext* context) const {
-  return nullptr;
```
Comment (goodov) [inline chromium_src/chrome/browser/safe_browsing/extension_telemetry/extension_telemetry_service_factory.cc:37]: `ProfileSelections::BuildNoProfilesSelected` prevents a service being constructor in any profile, so this is not needed.

**Review by cdesouza-chromium — APPROVED** [review]
Great clean up. LGTM.

File: chromium_src/chrome/browser/themes/theme_service_factory.cc (hunk 1 above)
Comment (simonhong) [reply chromium_src/chrome/browser/themes/theme_service_factory.cc:28]: Right, `BraveThemeService` doesn't have logic that check tor or not anymore.

**Review by simonhong — APPROVED** [review]
++

**Pull request merged**